### PR TITLE
add markup to templateUtility

### PR DESCRIPTION
**Pull request by superyyrrzz**

### CODE REVIEW

**Comment by hellosnow** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Build.Engine/TemplatePreprocessors/TemplateUtility.cs [hunk 1]
```diff
@@ -3,7 +3,11 @@
 
 namespace Microsoft.DocAsCode.Build.Engine
 {
+    using System;
+
     using Microsoft.DocAsCode.Common;
+    using Microsoft.DocAsCode.Plugins;
+    using Microsoft.DocAsCode.Exceptions;
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Build.Engine/TemplatePreprocessors/TemplateUtility.cs]: order

File: src/Microsoft.DocAsCode.Build.Engine/TemplatePreprocessors/TemplateUtility.cs [hunk 2]
```diff
@@ -54,5 +58,33 @@ public string GetHrefFromRoot(string originalHref, string sourceFileKey)
             }
             return file.UrlEncode().ToString() + UriUtility.GetQueryStringAndFragment(originalHref);
         }
+
+        public string Markup(string markdown, string sourceFileKey)
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Build.Engine/TemplatePreprocessors/TemplateUtility.cs]: sourceFileKey can be provided by us?

Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Build.Engine/TemplatePreprocessors/TemplateUtility.cs]: yes, added it to system metadata when adding templateUtility.ResolveSourceRelativePath

Comment (vicancy) [inline src/Microsoft.DocAsCode.Build.Engine/TemplatePreprocessors/TemplateUtility.cs]: so that no need to be a parameter to markup?

Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Build.Engine/TemplatePreprocessors/TemplateUtility.cs]: As it's hard to pass it here, so preserve as a parameter

---
In reply to: [113619942](https://github.com/dotnet/docfx/pull/1606#discussion_r113619942) [](ancestors = 113619942)

**Comment by superyyrrzz** [comment]
Seems this feature is needed before BUILD(5/10), it should be removed later as it breaks incremental build's design.

**Comment by vicancy** [comment]
Add a TODO if this is a temp solution

**Pull request merged**